src/ucp/wireup/select.c

```diff
@@ -653,22 +653,38 @@ static double ucp_wireup_fp8_pack_unpack_latency(double latency)
 }
 
 static inline double
-ucp_wireup_tl_iface_latency(ucp_context_h context,
-                            const uct_iface_attr_t *iface_attr,
+ucp_wireup_tl_iface_latency(const ucp_worker_iface_t *wiface,
                             const ucp_address_iface_attr_t *remote_iface_attr)
 {
-    double local_lat, lat_nsec;
+    ucp_context_h context = wiface->worker->context;
+    double local_lat;
 
     if (remote_iface_attr->addr_version == UCP_OBJECT_VERSION_V1) {
+        /* With the old protocols, iface latency/bw attributes are already
+         * updated based on topology distance. That is NOT the case with the
+         * new protocols for the local iface. Therefore, we need to add the
+         * distance latency for the local iface in here.
```

Comment:
this commend seems relevant for both if and else
maybe move it before the if/else?